### ISSUE TITLE
Setup up on 9.XX

### ISSUE THREAD

**Issue by superdale** [jonnyzzz/TeamCity.GitHub#83]
I have been working on getting this working since Monday. I have check setting though Teamcity and I am still getting this error. Could someone please help.

[2015-06-12 18:27:01,367]   WARN - .github.api.impl.GitHubApiImpl - Failed to complete query to GitHub with:
  requestURL: https://api.github.com/company/master/repos/username/https://api.github.com/company/master/statuses/a1ad9f5512e61ca2d3e8b5d81169ee81aa8b9dad
  requestMethod: POST
  requestEntity: {"state":"pending","target_url":"http://teamcity.com:8111/viewLog.html?buildId\u003d776\u0026buildTypeId\u003dmaster_masterBuild","description":"Started TeamCity Build Nuvis :: Nuvis Build","context":"continuous-integration/teamcity"}
  response: HTTP/1.1 404 Not Found
  responseEntity: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"} 
[2015-06-12 18:27:01,367]   WARN - lty.github.ChangeStatusUpdater - Failed to update GitHub status for hash: a1ad9f5512e61ca2d3e8b5d81169ee81aa8b9dad, buildId: 776, status: Pending. Failed to complete request to GitHub. Status: HTTP/1.1 404 Not Found 
java.io.IOException: Failed to complete request to GitHub. Status: HTTP/1.1 404 Not Found
        at jetbrains.teamcilty.github.api.impl.GitHubApiImpl.setChangeStatus(GitHubApiImpl.java:131)
        at jetbrains.teamcilty.github.ChangeStatusUpdater$1$1.run(ChangeStatusUpdater.java:272)
        at jetbrains.buildServer.util.ExceptionUtil$1.run(ExceptionUtil.java:40)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)

**Comment by jonnyzzz** [comment]
Please make sure you have enough permissions on GitHub tho complete the
request

Is the commit from the url exists?

**Comment by superdale** [comment]
We had this user working with 7.XX we just upgraded to 9.XX and cannot get it to work now.

**Comment by nuvibot** [comment]
I see what your talking about now. I have step through the steps and found the url that is getting block. Is this being blocked on Teamcity or github?

**Comment by superdale** [comment]
I did the github test and it shows I have access. ssh -T git@github.com. Should I use token instead of password. If so what token? I have tried the token from Teamcity server and the token from the github with no luck.

**Comment by superdale** [comment]
[2015-06-15 21:40:22,796]   WARN - .github.api.impl.GitHubApiImpl - Failed to complete query to GitHub with:
  requestURL: https://api.github.com/repos/Username/repos/statuses/5749bd2a177a0aa83a7de68b0296baabad013f9d
  requestMethod: POST
  requestEntity: {"state":"pending","target_url":"http://util.nuviapp.com:8111/viewLog.html?buildId\u003d907\u0026buildTypeId\u003dNuvis_Build","description":"Started TeamCity Build ::  Build"}
  response: HTTP/1.1 404 Not Found
  responseEntity: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"} 
[2015-06-15 21:40:22,796]   WARN - lty.github.ChangeStatusUpdater - Failed to update GitHub status for hash: 5749bd2a177a0aa83a7de68b0296baabad013f9d, buildId: 907, status: Pending. Failed to complete request to GitHub. Status: HTTP/1.1 404 Not Found 
java.io.IOException: Failed to complete request to GitHub. Status: HTTP/1.1 404 Not Found
        at jetbrains.teamcilty.github.api.impl.GitHubApiImpl.setChangeStatus(GitHubApiImpl.java:131)
        at jetbrains.teamcilty.github.ChangeStatusUpdater$1$1.run(ChangeStatusUpdater.java:272)
        at jetbrains.buildServer.util.ExceptionUtil$1.run(ExceptionUtil.java:40)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)

**Comment by jonnyzzz** [comment]
Do you see the commit 5749bd2a177a0aa83a7de68b0296baabad013f9d in you repo
or pull request?

If not, please check the rule you use for branches spec in the vcs root.
What rule are you using?

The ssh access does not sound for GitHub rest api. You should either use
username/password or access token that is generated from the GitHub.

BTW. Make sure you use the latest build if the plugin.

**Comment by superdale** [comment]
+:refs/pull/(*/merge)
+:refs/heads/*

I have the build pulling with username and password. teamcity.github I have set with username and password.

**Comment by elordahl** [comment]
Look at the URL in the error :
https://api.github.com/repos/Username/repos/statuses/5749bd2a177a0aa83a7de68b0296baabad013f9d

Is "Username" correct? Can you take a screenshot of your configuration in
TeamCity.

**Comment by jonnyzzz** [comment]
The
```
refs/pull/*/merge
```
Is the virtual commit GitHub creates for every pull request. The commit is
not shown in the ui and that looks like the reason http requests fail

You should use `refs/pull/*/head` instead

**Comment by superdale** [comment]
Thanks I will try it.

**Comment by superdale** [comment]
I am not using the default directory does that affect this plugin?

**Comment by superdale** [comment]
![screen shot 2015-06-17 at 9 09 36 am](https://cloud.githubusercontent.com/assets/12499885/8210865/3f7c2cf0-14d1-11e5-8d12-0e69a88cbe7f.png)

**Comment by superdale** [comment]
![screen shot 2015-06-17 at 9 15 28 am](https://cloud.githubusercontent.com/assets/12499885/8210939/aeec6bf4-14d1-11e5-9192-c672012a1587.png)

**Comment by superdale** [comment]
https://github.com/******/***** is the cvn root name

**Comment by superdale** [comment]
Used the curl command and came up with this error message. 

-bash-4.1$ curl --user "bot:password" -X POST https://api.github.com/company/repos/bot/master/statuses/0c87a8f7029c59b44bd15af547f1cad84f34bbf4 -d {"state":"success","target_url":"http://util.buildboxurl:8111/viewLog.html?buildId\u003d1105\u0026buildTypeId\u003dBuild","description":"Finished TeamCity Build master :: Build_master : Running"}
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}
curl: (6) Could not resolve host: target_url:http
curl: (6) Could not resolve host: description:Finished TeamCity Build master :: Build_master 
-bash-4.1$

**Comment by jonnyzzz** [comment]
Why have you changed the GitHub API URL in the build feature? You should
not change it if you use standard github.

Best regards,
Eugene Petrenko

**Comment by superdale** [comment]
I was trying anything to get this to work. Currently it is set to the default value.

**Comment by superdale** [comment]
Still not working.

**Comment by ajcurtin** [comment]
I am seeing this same issue.

**Comment by superdale** [comment]
I check the permission and the bot has full access to the repository. Any thoughts of what we can try next?

**Comment by ajcurtin** [comment]
Hi @superdale I fixed this issue for myself. Same 404 problem was happening. Turns out, the "owner" I had listed in the plugin configuration was not what the plugin wanted. I verified this by using curls against the github api until it returned the data I wanted. I'd make sure that the "owner" is what you think. 

For example: for me, my repo is /Companyname/Repo but I had the owner set to the github username of that repo, eg: Alex. I changed the Owner in the github plugin configuration to Companyname and everything worked again.

Hope this helps.
